### PR TITLE
syncer(dm): fix different output format for operate-schema get

### LINKED ISSUE #5688: different output format for operate-schema get
### What did you do?

https://github.com/pingcap/tiflow/blob/a1ae6e05df961d5218ff17959f3ad447fe05f146/dm/syncer/schema.go#L72-L84

when enter line 76, the output is like 

```
"CREATE TABLE `t_target` (\n  `id` bigint(20) NOT NULL,\n  `c1` varchar(20) DEFAULT NULL,\n  `c2` varchar(20) DEFAULT NULL,\n  PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */\n) ENGINE=InnoDB DEFAULT CHARSET=latin1 COLLATE=latin1_bin"
```
It contains `\n` and the table name is not the same as the operate-schema argument.

when enter line 82, the output is like 

```
" CREATE TABLE `t1` ( `id` bigint(20) NOT NULL, `c1` varchar(20) DEFAULT NULL, `c2` varchar(20) DEFAULT NULL, PRIMARY KEY (`id`) /*T![clustered_index] CLUSTERED */\n)) ENGINE=InnoDB DEFAULT CHARSET=latin1 COLLATE=latin1_bin"
```

introduced in https://github.com/pingcap/tiflow/pull/5350

### What did you expect to see?

same output format

### What did you see instead?

see above

### Versions of the cluster

master

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by ForwardStar**
### What problem does this PR solve?

Issue Number: close #5688 

### What is changed and how it works?

- If the checkpoint doesn't exist and the schema is retrieved from the downstream, replace the downstream table name with the requested upstream table name

- Delete all ``\n`` in the SQL results for the operate-schema result

#### Tests 

 - Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: dm/syncer/syncer_test.go [hunk 1]
```diff
@@ -1025,6 +1026,35 @@ func (s *testSyncerSuite) TestRun(c *C) {
...
+
+	syncer.checkpoint.(*RemoteCheckPoint).points = make(map[string]map[string]*binlogPoint)
+
+	mock.ExpectQuery("SHOW CREATE TABLE " + "`test_1`.`t_2`").WillReturnRows(
+		sqlmock.NewRows([]string{"Table", "Create Table"}).
+			AddRow("t_2", "CREATE TABLE `t_2` (\n `id` int(11) NOT NULL,\n `name` varchar(24) DEFAULT NULL,\n PRIMARY KEY (`id`) /*T![clustered_index] NONCLUSTERED */,\n KEY `index1` (`name`)\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin"))
+
+	sourceSchemaFromDownstream, err := syncer.OperateSchema(ctx, &pb.OperateWorkerSchemaRequest{Op: pb.SchemaOp_GetSchema, Database: "test_1", Table: "t_1"})
+	c.Assert(err, IsNil)
+	c.Assert(sourceSchemaFromCheckPoint, Equals, sourceSchemaFromDownstream)
```
[the remark this answers is not in the document]
Comment (lance6716) [reply dm/syncer/syncer_test.go:1060]: I mean the "expected" value is a string literal, not the "got" value, so we can make sure `sourceSchemaFromCheckPoint` and `sourceSchemaFromDownstream` is same and both equals a readable value.

[the remark this answers is not in the document]
Comment (dsdashun) [reply dm/syncer/syncer_test.go:1060]: Here we only compared that `sourceSchemaFromCheckPoint` and `sourceSchemaFromDownstream` are equal, but there are chances that both `sourceSchemaFromCheckPoint` and `sourceSchemaFromDownstream` are the same incorrect values.  ( For example, both of those values are an empty string, this test can pass).  You can define a string literal to specify the expected value, and compare the expected value with `sourceSchemaFromCheckPoint` or `sourceSchemaFromDownstream`.

**Review by lance6716 — APPROVED** [review]

**Review by dsdashun — APPROVED** [review]
LGTM

**Comment by ForwardStar** [comment]
/cherry-pick-release-5.4

**Pull request merged**